backend/app/status_manager.py

```
@@ -0,0 +1,203 @@
+    end_time: Optional[datetime] = None
+    error: Optional[str] = None
+    root_url: Optional[str] = None # Store the root URL for reference
+    data_extracted: Optional[str] = None # Total data extracted size (e.g., "11.02 KB")
+
+# In-memory storage for job statuses
+# NOTE: This is ephemeral and will be lost on server restart.
+# Consider a more persistent store (e.g., Redis, DB) for production.
+# Use the managed dictionary instead of the global one
+crawl_jobs: Dict[str, CrawlJobStatus] = crawl_jobs_managed # Type hint remains Dict for compatibility
+
+def initialize_job(job_id: str, root_url: str):
```

Ruff (0.8.2)

Missing return type annotation for public function `initialize_job`

Add return type annotation: `None`

(ANN201)